pkg/server/apiHandler.go

```diff
@@ -249,30 +156,13 @@ func (h *apiHandler) info(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
 		return
 	}
-	msg := &info.ZInfoMsg{}
-	if err := proto.Unmarshal(b, msg); err != nil {
-		log.Printf("Failed to parse info message: %v", err)
-		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
-		return
-	}
-	var entryBytes []byte
-	if entryBytes, err = protojson.Marshal(msg); err != nil {
-		log.Printf("Failed to marshal info message: %v", err)
-		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
-		return
-	}
-	select {
-	case h.infoChannel <- entryBytes:
-	default:
-	}
-	err = h.manager.WriteInfo(*u, entryBytes)
+	status, err := infoProcess(h.manager, h.infoChannel, *u, b)
```

Comment:
What do we gain by moving all of this to a single func `infoProcess()`, which only is used here?

Reply:
Not only here, but also inside apiHandlerV2